<div style="max-width: 700px; margin: 20px auto 30px; padding: 20px; text-align: center; border-bottom: 2px solid #e5e7eb;">
  <a href="02_data_cleaning.ipynb" style="display: inline-block; padding: 10px 24px; background: #f9fafb; color: #38549c; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; border: 1px solid #e5e7eb; margin-right: 12px;">
    ← Part 2 - Data Preparation
  </a>
  <a href="04_arma_models.ipynb" style="display: inline-block; padding: 10px 24px; background: linear-gradient(180deg, #38549c 0%, #14276c 100%); color: white; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; box-shadow: 0 4px 12px rgba(56,84,156,0.3);">
    Part 4 - ARMA Models →
  </a>
</div>

<div style="max-width: 880px; margin: 20px auto 22px; padding: 0px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06); overflow: hidden;">
  <div style="padding: 34px 32px 14px; text-align: center; line-height: 1.38;">
    <div style="font-size: 13px; letter-spacing: 0.14em; text-transform: uppercase; color: #6b7280; font-weight: bold; margin-bottom: 5px;">
      Session #1 • Part 3
    </div>
    <!-- <div style="font-size: 29px; font-weight: 800; color: #14276c; margin-bottom: 4px;">
      Linear Time Series Models
    </div> -->
    <div style="font-size: 29px; font-weight: 800; color: #14276c; margin-bottom: 4px;">
      Stationarity and Unit Root Tests
    </div>
    <div style="font-size: 16.5px; color: #374151; font-style: italic; margin-bottom: 0;">
      Advanced Training School: Methods for Time Series
    </div>
  </div>
  <div style="background: none; text-align: center; margin: 30px 0 10px;">
    <img src="https://www.cemfi.es/images/Logo-Azul.png" alt="CEMFI Logo" style="width: 158px; filter: drop-shadow(0 2px 12px rgba(56,84,156,0.05)); margin-bottom: 0;">
  </div>
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1.22em; font-weight: bold; margin-bottom: 0px;">
    Jesus Villota Miranda © 2025
  </div>
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1em; margin-top: 7px; margin-bottom: 20px;">
    <a href="mailto:jesus.villota@cemfi.edu.es" style="color: #38549c; text-decoration: none; margin-right:8px;">jesus.villota@cemfi.edu.es</a>
    <span style="color:#9fa7bd;">|</span>
    <a href="https://www.linkedin.com/in/jesusvillotamiranda/" target="_blank" style="color: #38549c; text-decoration: none; margin-left:7px;">LinkedIn</a>
  </div>
</div>

<div style="max-width: 700px; margin: 20px auto 30px; padding: 24px 32px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06);">
<!-- <div style="font-size: 14px; letter-spacing: 0.1em; text-transform: uppercase; color: #6b7280; font-weight: bold; margin-bottom: 12px; text-align: center;">
  Session #1 • Part 2
</div> -->
<div style="font-size: 22px; font-weight: 700; color: #14276c; margin-bottom: 16px; text-align: center;">
  Table of Contents
</div>
<div style="font-size: 15px; color: #374151; line-height: 1.8;">
<div style="margin-bottom: 12px;"><strong>1.</strong> <a href="#section-1" style="color: #38549c; text-decoration: none;">Introduction to Stationarity</a></div>
<div style="margin-left: 20px; margin-bottom: 8px;">1.1 <a href="#section-1-1" style="color: #38549c; text-decoration: none;">Spurious Regression Simulation</a></div>
<div style="margin-bottom: 12px;"><strong>2.</strong> <a href="#section-2" style="color: #38549c; text-decoration: none;">Unit Root Tests: Theory and Methods</a></div>
<div style="margin-bottom: 12px;"><strong>3.</strong> <a href="#section-3" style="color: #38549c; text-decoration: none;">Application to IBEX-35</a></div>
<div style="margin-left: 20px; margin-bottom: 8px;">3.1 <a href="#section-3-1" style="color: #38549c; text-decoration: none;">Time Series Plots</a></div>
<div style="margin-left: 20px; margin-bottom: 8px;">3.2 <a href="#section-3-2" style="color: #38549c; text-decoration: none;">Correlograms</a></div>
<div style="margin-left: 20px; margin-bottom: 8px;">3.3 <a href="#section-3-3" style="color: #38549c; text-decoration: none;">DF-GLS Tests</a></div>
<div style="margin-left: 20px; margin-bottom: 8px;">3.4 <a href="#section-3-4" style="color: #38549c; text-decoration: none;">Interpreting Test Output</a></div>
</div></div>

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-1"></a>
<h2 class="styled-header">1) Introduction to Stationarity</h2>

**Stationarity** is a fundamental concept in time series analysis. A time series is **stationary** if its statistical properties (mean, variance, autocovariance) do not change over time.

**Why Stationarity Matters**:
- Standard statistical inference requires stationarity
- Non-stationary series can lead to **spurious regressions**
- Many time series models (ARMA) assume stationarity
- Non-stationary series often need transformation (differencing) before modeling

**Types of Stationarity**:

1. **Strict Stationarity**: Joint distribution invariant to time shifts
2. **Weak (Covariance) Stationarity**: 
   - Constant mean: $\mathbb{E}[Y_t] = \mu$ for all $t$
   - Constant variance: $\mathbb{V}[Y_t] = \sigma^2$ for all $t$
   - Autocovariance depends only on lag: $\gamma(t, t-h) = \gamma(h)$

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-1-1"></a>
### 1.1 Spurious Regression Simulation

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 20px; border-radius: 8px; margin: 20px 0;">

**The Spurious Regression Problem**: One of the most important discoveries in econometrics (Granger & Newbold, 1974) is that regressing one non-stationary series on another can produce **highly misleading results** even when the variables are completely independent.

</div>

#### Mathematical Framework

We construct two **independent random walks** with no causal relationship whatsoever:

$$
\begin{align*}
x_t &= x_{t-1} + \varepsilon_t, \quad x_0 = 0 \\
y_t &= y_{t-1} + \nu_t, \quad y_0 = 0
\end{align*}
$$

where $\varepsilon_t, \nu_t \stackrel{\text{i.i.d.}}{\sim} \mathcal{N}(0,1)$ are **independent** standard normal shocks.

**Key Properties of Random Walks**:
1. **Non-stationary**: $\mathbb{E}[x_t] = 0$ but $\text{Var}(x_t) = t\sigma^2 \to \infty$ as $t \to \infty$
2. **No mean reversion**: Shocks have permanent effects
3. **Trending behavior**: Series wander away from their starting values

#### The Spurious Regression

Despite being generated independently, when we estimate:

$$y_t = \alpha + \beta x_t + u_t$$

we typically observe:
- **Inflated $R^2$** (often 0.2-0.8, depending on the random sample) suggesting fit where none exists
- **Significant t-statistics** (often |t| > 2) suggesting "significant" relationship
- **Low Durbin-Watson statistic** (DW ≈ 0) indicating strongly serially correlated residuals

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Why This Happens**: Both series drift randomly over time. By chance, they may drift in similar directions during certain periods, creating the **illusion** of a relationship. The residuals $\hat{u}_t$ inherit the non-stationarity of $x_t$ and $y_t$, violating the basic assumption that errors are stationary with constant variance.

</div>

#### Simulation Objectives

1. Generate two independent random walks ($T = 250$ observations)
2. Run OLS regression: $y_t = \alpha + \beta x_t + u_t$
3. Document the spurious results (high $R^2$, significant $t$-stat)
4. Visualize the false relationship and non-stationary residuals

In [14]:
* ---------------------------------------------------------------------------
* Spurious Regression Demonstration: Two Independent Random Walks
* ---------------------------------------------------------------------------

clear all
set obs 250                    // Sample size (250 observations)
set seed 12345                 // Seed that produces more typical spurious result

* Generate time index and declare as time series
gen t = _n
tsset t                        // Declare time series structure

* Generate independent random shocks: ε_t, ν_t ~ N(0,1)
gen eps = rnormal()
gen nu = rnormal()

* Construct random walks: x_t = x_{t-1} + ε_t, y_t = y_{t-1} + ν_t
gen x_rw = sum(eps)            // Cumulative sum creates random walk
gen y_rw = sum(nu)             // Completely independent of x_rw

* Display summary statistics for the random walks
display _newline "{txt}{hline 60}"
display "{txt}Summary Statistics for Independent Random Walks"
display "{txt}{hline 60}"
summarize x_rw y_rw

* Correlation between contemporaneous values (spurious!)
display _newline "{txt}Correlation between independent random walks:"
correlate x_rw y_rw

* Run the spurious regression: y_t = α + β·x_t + u_t
display _newline "{txt}{hline 60}"
display "{txt}Spurious Regression: y_t = α + β·x_t + u_t"
display "{txt}{hline 60}"
quietly regress y_rw x_rw

* Calculate Durbin-Watson statistic explicitly
quietly estat dwatson
local dw = r(dw)

* Extract key statistics
local beta = _b[x_rw]
local se_beta = _se[x_rw]
local t_stat = `beta'/`se_beta'
local r2 = e(r2)
local N = e(N)

* Display full regression output
regress y_rw x_rw

* Display interpretation
display _newline "{txt}{hline 60}"
display "{txt}Spurious Regression Results Summary:"
display "{txt}{hline 60}"
display "{txt}Coefficient (β):     {res}" %7.4f `beta' "{txt}  (should be ~0 if no relation)"
display "{txt}Standard Error:      {res}" %7.4f `se_beta'
display "{txt}t-statistic:         {res}" %7.3f `t_stat' "{txt}  (appears 'significant'!)"
display "{txt}R-squared:           {res}" %7.4f `r2' "{txt}  (" %4.1f `r2'*100 "% of variation 'explained')"
display "{txt}Durbin-Watson:       {res}" %7.4f `dw' "{txt}  (DW≈0 indicates strong serial correlation)"
display "{txt}{hline 60}"
display "{err}⚠  WARNING: Despite statistical significance,"
display "{err}   these series are INDEPENDENT by construction!"
display "{err}   This demonstrates the SPURIOUS REGRESSION problem."
display "{txt}{hline 60}" _newline

* Generate predictions and residuals for visualization
predict yhat, xb
predict resid, resid

* VISUALIZATION 1: Time series plots of both random walks
twoway (line x_rw t, lcolor(navy) lwidth(medium)) ///
       (line y_rw t, lcolor(maroon) lwidth(medium)), ///
    title("Two Independent Random Walks", size(medium) color(black)) ///
    subtitle("Both series drift randomly with no causal link", size(small)) ///
    ytitle("Value", size(small)) xtitle("Time", size(small)) ///
    legend(order(1 "x{subscript:t} (random walk 1)" 2 "y{subscript:t} (random walk 2)") ///
           size(small) position(6) rows(1)) ///
    name(sr_timeseries, replace)

* VISUALIZATION 2: Scatter plot with spurious regression line
twoway (scatter y_rw x_rw, msymbol(circle) msize(small) mcolor(navy%40)) ///
       (lfit y_rw x_rw, lcolor(red) lwidth(medthick)), ///
    title("Spurious Regression: y{subscript:t} vs x{subscript:t}", size(medium) color(black)) ///
    subtitle("R² = `: display %4.3f `r2'' with t = `: display %4.2f `t_stat'' despite independence", size(small)) ///
    ytitle("y{subscript:t} (random walk 2)", size(small)) ///
    xtitle("x{subscript:t} (random walk 1)", size(small)) ///
    legend(order(1 "Observed pairs" 2 "OLS fit: β = `: display %5.3f `beta''") ///
           size(small) position(6) rows(1)) ///
    note("False relationship due to both series being non-stationary", size(vsmall)) ///
    name(sr_scatter, replace)

* VISUALIZATION 3: Residual plot showing non-stationary behavior
twoway (line resid t, lcolor(red) lwidth(medium)) ///
       (line yhat t, lcolor(blue%30) lwidth(thin) lpattern(dash)), ///
    title("Non-Stationary Residuals from Spurious Regression", size(medium) color(black)) ///
    subtitle("Residuals drift and exhibit strong persistence (DW = `: display %4.3f `dw'')", size(small)) ///
    ytitle("Value", size(small)) xtitle("Time", size(small)) ///
    yline(0, lcolor(gs10) lpattern(solid)) ///
    legend(order(1 "Residuals (û{subscript:t})" 2 "Fitted values (ŷ{subscript:t})") ///
           size(small) position(6) rows(1)) ///
    name(sr_resid, replace)

* VISUALIZATION 4: ACF of residuals (should show strong persistence)
ac resid, lags(40) ///
    title("ACF of Residuals: Evidence of Non-Stationarity", size(medium)) ///
    subtitle("Slow decay indicates residuals inherit the unit root", size(small)) ///
    name(sr_acf_resid, replace)



Number of observations (_N) was 0, now 250.




Time variable: t, 1 to 250




Time variable: t, 1 to 250
        Delta: 1 unit






------------------------------------------------------------

Summary Statistics for Independent Random Walks

------------------------------------------------------------


    Variable |        Obs        Mean    Std. dev.       Min        Max
-------------+---------------------------------------------------------
        x_rw |        250    13.16363    5.139517    .957395   25.76464
        y_rw |        250    4.826374    4.171077   -5.33087   14.42246


Correlation between independent random walks:

(obs=250)

             |     x_rw     y_rw
-------------+------------------
        x_rw |   1.0000
        y_rw |  -0.2337   1.0000



------------------------------------------------------------

Spurious Regression: y_t = α + β·x_t + u_t

------------------------------------------------------------
        Delta: 1 unit






---------------------

<link rel="stylesheet" href="notebook_styles.css">

#### Interpretation of Results

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Simulation Reveals:**

1. **Time Series Plot** (`sr_timeseries`): Shows both random walks drifting independently. Any visual similarity is purely coincidental.

2. **Scatter Plot** (`sr_scatter`): Despite independence, we observe:
   - A "significant" regression line with slope $\hat{\beta}$
   - Inflated $R^2$ suggesting the model explains variation
   - Large t-statistic (often |t| > 2) suggesting $\beta \neq 0$ with high "confidence"
   
3. **Residual Plot** (`sr_resid`): **This is the smoking gun**
   - Residuals $\hat{u}_t$ exhibit **persistent wandering** rather than random fluctuation
   - Durbin-Watson statistic near 0 (far below 2) confirms strong serial correlation
   - The residuals themselves behave like a non-stationary process

4. **ACF of Residuals** (`sr_acf_resid`): Shows **slow decay** characteristic of unit root processes
   - In valid regression, ACF should cut off quickly
   - Here, autocorrelations remain large and significant at long lags

</div>

#### The Fundamental Lesson

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Standard OLS inference requires stationary regressors.** When both $x_t$ and $y_t$ are integrated of order 1 (denoted $I(1)$):

- **t-statistics do not follow t-distributions** → significance tests are invalid
- **$R^2$ is inflated** → goodness of fit is meaningless
- **Standard errors are wrong** → confidence intervals are incorrect

**Solutions**:
1. **Difference the data**: Work with $\Delta y_t$ and $\Delta x_t$ (returns rather than levels)
2. **Test for cointegration**: If a linear combination $y_t - \beta x_t$ is $I(0)$, use cointegration methods
3. **Unit root tests**: Always test for stationarity before running regressions with time series

</div>

**Historical Note**: Granger and Newbold (1974) found that in Monte Carlo simulations of spurious regressions, the probability of falsely rejecting the null of no relationship exceeded **75%** at the 5% significance level—far worse than the nominal 5% Type I error rate.

<div style="background: #fef2f2; border: 2px solid #dc2626; padding: 14px; border-radius: 8px; margin: 16px 0; text-align: center;">
<strong>⚠️ Bottom Line:</strong> Never trust regression results with non-stationary time series without first testing for unit roots or establishing cointegration.
</div>

<link rel="stylesheet" href="notebook_styles.css">

Having seen the danger of spurious regression, we now learn formal methods to test for stationarity.

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-2"></a>
<h2 class="styled-header">2) Unit Root Tests: Theory and Methods</h2>

Before implementing unit root tests, we need to understand the mathematical framework underlying them. We'll first introduce the **Augmented Dickey-Fuller (ADF)** test, then the more powerful **DF-GLS** test.

#### The Augmented Dickey-Fuller (ADF) Test

The ADF test checks whether a time series has a unit root by estimating the following regression:

$$\Delta Y_t = \alpha + \delta Y_{t-1} + \sum_{i=1}^{p} \beta_i \Delta Y_{t-i} + \varepsilon_t$$

where:
- $\Delta Y_t = Y_t - Y_{t-1}$ is the first difference
- $\alpha$ is a constant (drift term)
- $\delta$ measures the degree of mean reversion
- The augmentation terms $\sum_{i=1}^{p} \beta_i \Delta Y_{t-i}$ account for serial correlation
- $\varepsilon_t$ is white noise

**Hypothesis Test**:
- **$H_0: \delta = 0$** (unit root exists → series is non-stationary)
- **$H_1: \delta < 0$** (series is stationary)

**Why $\delta < 0$ indicates stationarity**: If $\delta < 0$, the series exhibits **mean reversion** — when $Y_{t-1}$ is above its long-run mean, $\Delta Y_t$ tends to be negative (pulling it back down), and vice versa.

**The Non-Standard Distribution Problem**: Under the null hypothesis, the test statistic $t_\delta = \hat{\delta}/\text{SE}(\hat{\delta})$ does **not** follow a standard t-distribution. Instead, it follows the **Dickey-Fuller distribution**, which is left-skewed with critical values that are more negative than the standard normal (e.g., -2.86 at 5% vs. -1.96 for standard normal). This is why we need special critical value tables.

#### The DF-GLS Test (Elliott-Rothenberg-Stock, 1996)

The **DF-GLS test** improves upon the ADF test by using **Generalized Least Squares (GLS) detrending** before performing the unit root test. This approach has substantially better power, especially when the series is close to having a unit root.

**The GLS Transformation**:

Instead of testing the original series $Y_t$, DF-GLS first removes deterministic components (constant and/or trend) using GLS. The transformed series is:

$$\tilde{Y}_t = \begin{cases}
Y_1 & \text{if } t = 1 \\
Y_t - \bar{c} Y_{t-1} & \text{if } t > 1
\end{cases}$$

where $\bar{c} = 1 - \frac{7}{T}$ for the constant-only case (the `notrend` option), or $\bar{c} = 1 - \frac{13.5}{T}$ when including a trend. This transformation gives more weight to early observations where the series behavior is most informative about stationarity.

**After GLS detrending**, the DF-GLS test runs the ADF regression on the transformed series:

$$\Delta \tilde{Y}_t = \delta \tilde{Y}_{t-1} + \sum_{i=1}^{p} \beta_i \Delta \tilde{Y}_{t-i} + \varepsilon_t$$

The test statistic is:

$$\text{DF-GLS} = \frac{\hat{\delta}}{\text{SE}(\hat{\delta})}$$

**Key Advantages**:
- **Higher power**: Better at detecting stationarity when $\delta$ is close to zero (near unit root)
- **Less size distortion**: Fewer false rejections when the null is true
- **Asymptotically equivalent** to ADF but with better finite-sample properties
- **Standard choice** in modern applied econometrics

**Connection to Stata Implementation**:
- `dfgls varname, notrend`: Uses $\bar{c} = 1 - \frac{7}{T}$ (constant-only detrending)
- `dfgls varname`: Uses $\bar{c} = 1 - \frac{13.5}{T}$ (constant + trend detrending)
- `maxlag(p)`: Specifies maximum augmentation lags in $\sum_{i=1}^{p} \beta_i \Delta \tilde{Y}_{t-i}$

<link rel="stylesheet" href="notebook_styles.css">

With the theoretical foundation in place, we now apply these tests to real financial data.


<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3"></a>
<h2 class="styled-header">3) Application to IBEX-35 Index and Returns</h2>

We test for stationarity in IBEX-35 data by comparing:

1. **Index levels** (prices) - expected to be **non-stationary**
2. **Returns** (first differences of log prices) - expected to be **stationary**


<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3-1"></a>
### 3.1 Time Series Plots: Levels vs Returns


<link rel="stylesheet" href="notebook_styles.css">

With the theoretical foundation in place, we now apply these tests to real financial data.

<a id="section-3"></a>
<h2 class="styled-header">3) Application to IBEX-35 Index and Returns</h2>

We test for stationarity in IBEX-35 data by comparing:

1. **Index levels** (prices) - expected to be **non-stationary**
2. **Returns** (first differences of log prices) - expected to be **stationary**

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3-1"></a>
### 3.1 Time Series Plots: Levels vs Returns

In [ ]:
* ----------------------------------------------------------------------------
* Application to IBEX-35 data
* ----------------------------------------------------------------------------

* Load IBEX-35 daily data from processed file
global processed_data "../../data/processed"  // Define path to processed data folder
use "$processed_data/ibex35_daily.dta", clear  // Load IBEX-35 daily dataset

* Time series plots
* Plot IBEX-35 closing prices over time
tsline ibex35_close, ///
    title("IBEX-35 Index Levels") ///
    ytitle("Index") xtitle("Date") ///
    name(ibex_levels, replace)

* Plot IBEX-35 returns over time
tsline ribex35, ///
    title("IBEX-35 Daily Returns") ///
    ytitle("Returns (%)") xtitle("Date") ///
    name(ibex_returns, replace)





file /Users/jesusvillotamiranda/.stata_kernel_cache/graph18.svg saved as SVG
    format
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph18.pdf saved as PDF
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph18.pdf saved as PDF
    format


file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.svg saved as SVG
    format


file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.pdf saved as PDF
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.pdf saved as PDF
    format


**Visual Inspection**:
- **Levels**: Show a clear **trend** (non-stationary - mean changes over time)
- **Returns**: Fluctuate around **zero** (stationary - constant mean)

This visual inspection suggests that levels are non-stationary while returns are stationary.

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3-2"></a>
### 3.2 Correlograms: Levels vs Returns

In [12]:
* Correlograms
* Plot autocorrelation function for IBEX-35 levels up to lag 40
ac ibex35_close, lags(40) ///
    title("ACF: IBEX-35 Levels") ///
    name(acf_ibex_levels, replace)

* Plot autocorrelation function for IBEX-35 returns up to lag 40
ac ribex35, lags(40) ///
    title("ACF: IBEX-35 Returns") ///
    name(acf_ibex_ret, replace)


(note: time series has 1752 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.pdf saved as PDF
    format

(note: time series has 1700 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.pdf saved as PDF
    format

(note: time series has 1700 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.pdf saved as PDF
    format

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.pdf saved as PDF
    format


**ACF Patterns**:
- **Levels**: ACF decays **very slowly** (characteristic of non-stationary series)
- **Returns**: ACF decays **quickly** and stays within confidence bands (characteristic of stationary series)

The slow decay in levels indicates **long memory** and suggests a unit root process.

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3-3"></a>
### 3.3 DF-GLS Unit Root Tests: Implementation

In [13]:
* DF-GLS unit root tests
* Create continuous series for dfgls (requires no gaps)
* Clear any previous preserve state that may have been left incomplete
capture restore  // Silently restore if any preserve state exists
preserve  // Save current dataset state to restore later
drop if missing(ibex35_close)  // Remove observations with missing IBEX-35 closing prices
gen t = _n  // Generate continuous time index from 1 to N
tsset t  // Declare dataset as time series with variable t as time index
display _newline "DF-GLS Test for IBEX-35 Levels:"  // Print header for test results
display "{hline 60}"
dfgls ibex35_close, maxlag(3) notrend  // Run DF-GLS unit root test (display full output)
display _newline "Interpretation:"
display "  H0: Series has a unit root (non-stationary)"
display "  If test statistic < critical value → reject H0 (stationary)"
display "  If test statistic > critical value → fail to reject H0 (non-stationary)"
restore  // Restore dataset to state before preserve

preserve  // Save current dataset state to restore later
drop if missing(ribex35)  // Remove observations with missing IBEX-35 returns
gen t = _n  // Generate continuous time index from 1 to N
tsset t  // Declare dataset as time series with variable t as time index
display _newline _newline "DF-GLS Test for IBEX-35 Returns:"  // Print header for test results
display "{hline 60}"
dfgls ribex35, maxlag(3) notrend  // Run DF-GLS unit root test (display full output)
display _newline "Interpretation:"
display "  Returns should show negative test statistics well below critical values"
display "  → Reject H0 → Returns are stationary"
restore  // Restore dataset to state before preserve





(0 observations deleted)



Time variable: t, 1 to 8197
        Delta: 1 unit


DF-GLS Test for IBEX-35 Levels:

------------------------------------------------------------



Time variable: t, 1 to 8197
        Delta: 1 unit


DF-GLS Test for IBEX-35 Levels:

------------------------------------------------------------


DF-GLS test for unit root                Number of obs = 8,193
Variable: ibex35_close
Lag selection: User specified            Maximum lag   =     3

                             -------- Critical value ---------


DF-GLS test for unit root                Number of obs = 8,193
Variable: ibex35_close
Lag selection: User specified            Maximum lag   =     3

                             -------- Critical value ---------
 [lags]       DF-GLS mu           1%           5%          10%
--------------------------------------------------------------
      3           0.440       -2.580       -1.950       -1.626
 [lags]       DF-GLS mu           1%           5%      

<link rel="stylesheet" href="notebook_styles.css">

<a id="section-3-4"></a>
### 3.4 Understanding DF-GLS Test Output

The DF-GLS test output follows a simple logic. Here's how to interpret it:

**Decision Rule**: In unit root tests, we reject when the test statistic is large and negative.

$$\text{Decision Rule: If } |\text{Test Statistic}| > |\text{Critical Value}| \text{ → Reject } H_0$$

| Outcome | Test Statistic | Decision | Interpretation |
|---------|----------------|----------|-----------------|
| **Very negative** | e.g., -29.924 | Reject $H_0$ | **Stationary** ✓ |
| **Less negative** | e.g., -2.5 | Reject or borderline | Depends on significance level |
| **Close to zero/positive** | e.g., 0.440 | Fail to reject $H_0$ | **Non-stationary** ✗ |

**Reading the Critical Values Table**:

```
                         -------- Critical value ---------
 [lags]       DF-GLS mu       1%           5%          10%
--------------------------------------------------------------
      3           0.440     -2.580       -1.950       -1.626
```

These three critical values correspond to different significance levels (1%, 5%, 10%). If your test statistic is **more negative** than the critical value, you reject the null hypothesis at that significance level.

**IBEX-35 Levels**: Test Statistic = **0.440**
- 0.440 > -2.580, -1.950, -1.626 (all critical values)
- **Interpretation**: At **all significance levels**, we **fail to reject** $H_0$. The series is **non-stationary**.

**IBEX-35 Returns**: Test Statistic = **-29.924**
- -29.924 < -2.580, -1.950, -1.626 (all critical values)
- **Interpretation**: At **all significance levels**, we **reject** $H_0$. The series is **stationary**.

**Lag Selection**: The output shows information criteria (Opt lag, Min SIC, Min MAIC) that help validate the choice of `maxlag(3)`. The `maxlag` parameter tells Stata to test specifications with 1, 2, and 3 lags to account for serial correlation in residuals.

**The `notrend` Option**: For financial data like stock prices, we use `notrend` because prices are driven by random shocks (stochastic trend), not a fixed linear deterministic trend.

**Key Insight**: The dramatic difference between levels and returns demonstrates the **power of differencing**:

$$\text{Price levels: } P_t \text{ (non-stationary, } I(1)) \quad \text{Returns: } r_t = \Delta \ln(P_t) \text{ (stationary, } I(0))$$

This transformation is crucial because standard regression assumes stationarity, and non-stationary regressors can cause spurious correlations.